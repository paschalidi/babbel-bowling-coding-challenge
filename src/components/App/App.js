import React from 'react';
import { connect } from 'react-redux';
import InitialGamePanel from '../InitialGamePanel';


class App extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    if (this.props.show === 'InitialGamePanel')
      return <InitialGamePanel />;
  }
}

function mapStateToProps(state) {
  return { ...state.game };
}

export default connect(mapStateToProps)(App);
