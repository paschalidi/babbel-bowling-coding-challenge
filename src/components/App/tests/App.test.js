import React from 'react';
import { Provider } from 'react-redux';
import App from '../App';
import renderer from 'react-test-renderer';

import Store from '../../../store/configure';
const StoreInstance = Store();

test('App renders', () => {
  const component = renderer.create(
    <Provider store={StoreInstance}>
      <App />
    </Provider>
  );
  let tree;

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
