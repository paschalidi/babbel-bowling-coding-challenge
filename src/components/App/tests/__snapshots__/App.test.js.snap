// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="flex panel background-color-black"
>
  <div
    className="ui centered container grid"
  >
    <div
      className="row"
      style={
        Object {
          "paddingBottom": "5vw",
        }
      }
    >
      <div
        className="glowing-text text-lg"
      >
        Welcome to Babbel Bowling!
      </div>
    </div>
    <div
      className="row"
    >
      <button
        className="ui purple inverted button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        New Game
      </button>
    </div>
    <div
      className="row"
    >
      <a
        href="https://www.thoughtco.com/bowling-scoring-420895"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          className="ui purple inverted button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Rules
        </button>
      </a>
    </div>
  </div>
</div>
`;
